corehq/apps/hqcase/views.py

```diff
@@ -79,13 +84,33 @@ def delete_cases(self, request, domain):
 @require_superuser_or_contractor
 @requires_privilege_with_fallback(privileges.API_ACCESS)
 def case_api(request, domain, case_id=None):
+    if request.method == 'GET' and case_id:
+        return _handle_individual_get(request, case_id)
+    if request.method == 'GET' and not case_id and domain_needs_search_index(domain):
+        return _handle_list_view(request)
     if request.method == 'POST' and not case_id:
         return _handle_case_update(request)
     if request.method == 'PUT' and case_id:
         return _handle_case_update(request, case_id)
     return JsonResponse({'error': "Request method not allowed"}, status=405)
 
 
+def _handle_individual_get(request, case_id):
+    try:
+        case = CaseAccessors(request.domain).get_case(case_id)
```

Comment:
I don't think this validates the domain.
You can do
```python
CaseAccessors(DomainX). get_case(case_id_from_domain_y)
```
and still get a result.

Reply:
What do you think of adding a test that tries to get a case from a different domain?

Reply:
Wow, I didn't realize that.  Yup, corrected and tested.

Reply:
Along these lines, I assume it is not possible to create or update a case if the case ID exists in some other domain. Do we have any tests for that?

Reply:
That's right, it's not possible.  You can't provide your own case IDs on creation, so it's a moot point there, and on update, the code checks that new dbaccessor `get_case_ids_that_exist`, which _does_ validate domain.  Just added a test demonstrating this failure.